app/models/setting.rb

```diff
@@ -106,12 +106,8 @@ def default=(v)
   def parse_string_value(val)
     case settings_type
     when "boolean"
-      val = val.downcase
-      if val == "true"
-        self.value = true
-      elsif val == "false"
-        self.value = false
-      else
+      self.value = Foreman::Cast.to_bool(val)
+      if self.value == self.default
```

Comment:
Nope, because of this https://github.com/theforeman/foreman/blob/develop/app/models/setting.rb#L92-92.  The value of `self.value` is never nil in fact.